File: examples/node/backdrop_demo.js
/**
 * winpane backdrop demo: Mica and Acrylic backdrop effects from Node.js.
 *
 * Creates a panel with Mica backdrop, waits 3 seconds, switches to Acrylic,
 * waits 3 seconds, then fades out.
 *
 * Requires Windows 11 22H2+ for backdrop effects.
 *
 * Usage:
 *   node backdrop_demo.js
 *
 * Requires: npm install winpane (or link to local build)
 */
// ── winpane design tokens ──────────────────────────────────────
// Surface base: #121216  Glass: +e4  Solid: +ff  Muted: +f2
// Elevated:     #1c1c21  Interactive: #26262cff  Hover: #303038ff
// Border:       #ffffff12  Text:      #e8e8edff   Muted: #9494a0ff
// Accent:       #528bffff  Success:   #34d399ff   Warning: #fbbf24ff
// Danger:       #ef4444ff  Radius: 10/6 px
// ────────────────────────────────────────────────────────────────
const { WinPane } = require('winpane');

const wp = new WinPane();
const panel = wp.createPanel({ width: 400, height: 300, monitor: 0, anchor: 'top_left', margin: 40 });

wp.setBackdrop(panel, 'mica');
wp.setText(panel, 'title', {
  text: 'Mica Backdrop from Node.js',
  x: 20, y: 20, fontSize: 16, color: '#e8e8ed',
});
wp.show(panel);

console.log('Showing Mica backdrop. Switching to Acrylic in 3 seconds...');

setTimeout(() => {
  wp.setBackdrop(panel, 'acrylic');
  wp.setText(panel, 'title', {
    text: 'Switched to Acrylic',
    x: 20, y: 20, fontSize: 16, color: '#e8e8ed',
  });
  console.log('Switched to Acrylic. Fading out in 3 seconds...');
}, 3000);

setTimeout(() => {
  console.log('Fading out...');
  wp.fadeOut(panel, 500);
}, 6000);

setTimeout(() => {
  wp.close();
  console.log('Done!');
}, 8000);

File: examples/rust/backdrop_demo.rs
//! Demo: Mica and Acrylic backdrop effects
//!
//! Creates two panels side by side: one with Mica backdrop, one with Acrylic.
//! Each has a title label with white text over the transparent backdrop.
//! Stays visible for 10 seconds.
//!
//! Requires Windows 11 22H2+ (build 22621) for backdrop effects.
//! On older versions, the panels appear with no backdrop (transparent).
//!
//! Run on Windows: cargo run -p winpane --example backdrop_demo

use winpane::{Backdrop, Color, Context, PanelConfig, TextElement};

#[allow(clippy::print_stdout)]
fn main() -> Result<(), winpane::Error> {
    let ctx = Context::new()?;

    // Left panel: Mica backdrop
    let panel1 = ctx.create_panel(PanelConfig {
        x: 100,
        y: 100,
        width: 300,
        height: 200,
        ..Default::default()
    })?;
    panel1.set_backdrop(Backdrop::Mica);
    panel1.set_text(
        "title",
        TextElement {
            text: "Mica Backdrop".into(),
            x: 20.0,
            y: 20.0,
            font_size: 24.0,
            color: Color::WHITE,
            ..Default::default()
        },
    );
    panel1.show();

    // Right panel: Acrylic backdrop
    let panel2 = ctx.create_panel(PanelConfig {
        x: 450,
        y: 100,
        width: 300,
        height: 200,
        ..Default::default()
    })?;
    panel2.set_backdrop(Backdrop::Acrylic);
    panel2.set_text(
        "title",
        TextElement {
            text: "Acrylic Backdrop".into(),
            x: 20.0,
            y: 20.0,
            font_size: 24.0,
            color: Color::WHITE,
            ..Default::default()
        },
    );
    panel2.show();

    println!("winpane backdrop_demo: two panels visible at (100,100) and (450,100).");
    println!("Left = Mica, Right = Acrylic. Closing in 10 seconds.");

    std::thread::sleep(std::time::Duration::from_secs(10));
    Ok(())
}
